classes/Snippet.php

```diff
@@ -464,7 +438,41 @@ protected static function dropDownOptionsToString($optionsArray)
         return implode(' | ', $result);
     }
 
-    protected static function extractSnippetsFromMarkup($markup, $theme)
+    /**
+     * Parse content to render snippets
+     *
+     * @throws ApplicationException
+     * @throws CmsException
+     * @throws SystemException
```

Comment:
Needs reasons why each exception would be thrown

Reply:
updated